### PR TITLE
Automatic prod deployments

### PR DESCRIPTION
**Pull request by r-czajkowski**
Closes: #7 

This PR adds a GH workflow that deploys the preview to `preview.threshold.network` on every pull request that targets to `main` branch and deploys website to `threshold.network` on push to `main` branch.

### CODE REVIEW

**Comment by pdyraga** [comment]
Can we clear the `preview.threshold.network` bucket once the previewed PR is merged? We do not have to block this PR but I think we should have this functionality both in this repo and in `threshold-network/token-dashboard`. We do not want to leave dangling, old code under the official domain. It is not helpful for SEO and can lead to security issues in case some scammer points someone to an old dashboard preview link.

File: .github/workflows/ci.yaml
Comment (nkuba) [inline .github/workflows/ci.yaml:58]:
```suggestion
          service-key: ${{ secrets.PROD_PREVIEW_UPLOADER_SERVICE_KEY_JSON_BASE64 }}
          project: ${{ secrets.PROD_PREVIEW_GOOGLE_PROJECT_ID }}
```

File: .github/workflows/ci.yaml
Comment (nkuba) [reply .github/workflows/ci.yaml:77]: The environment was configured for this repository, so we can use it now:
```suggestion
    if: github.event_name == 'push'
    # production environment is protected, it requires an approval before execution.
    environment:
      name: prod
```

**Review by nkuba — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml
Comment (r-czajkowski) [reply .github/workflows/ci.yaml]: [97de80b](https://github.com/threshold-network/splash-page/pull/8/commits/97de80bd6453435f3a6c3665231fa999ce15bf67)

**Review by nkuba — APPROVED** [review]
Looks like it worked 💯

**Pull request merged**
